src/repository/dotRepository.js

```diff
@@ -13,19 +19,40 @@ async function getId(scriptId, actorId) {
   return result.rows[0].id;
 }
 
+/**
+ * Update the position of the progress dot
+ * @param {number} dotId ID of the current progress dot
+ * @param {string} position New position of the progress dot
+ */
 async function set(dotId, position) {
   await db.query(`UPDATE read_position SET position = $2 WHERE id = $1`, [dotId, position]);
 }
 
+/**
+ * Get all the progress dots for the selected script
+ * @param {string} scriptId ID of the selected script
+ * @return {Promise<Array.<Dot>>}
+ */
 async function getAll(scriptId) {
   const result = await db.query('SELECT id, actor_id, position FROM read_position WHERE script_id = $1', [scriptId]);
 
-  return result.rows.map((dot) => ({
-    id: dot.id,
-    actorId: dot.actor_id,
-    scriptId: scriptId,
-    position: dot.position,
-  }));
+  return result.rows.map((dot) => (new Dot(dot.id, dot.actor_id, scriptId, dot.position)));
+}
+
+/** Dot Class */
```

Comment:
I'm a bit surprised this is necessary when it just has the exact same information as the next line of code

Reply:
Perhaps it isn't necessary. I felt like it allowed for clearer type documentation of what the function was returning. Do you think the cons of the single-use class are greater than the pros of the type clarity? I'm open to reverting it.

Reply:
Per our conversation, I misunderstood your comment and have removed the redundant '{name} class'